src/System.Text.CaseFolding/System/Text/CaseFolding/SimpleCaseFoldingMapSurrogates.cs

```diff
@@ -0,0 +1,135 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.CaseFolding
+{
+    /// <summary>
+    /// Simple case folding methods.
+    /// </summary>
+    public static partial class SimpleCaseFolding
+
+    {
+        private static readonly ushort[] MapSurrogateLevel1 =
```

Comment:
>MapSurrogateLevel1 [](start = 41, length = 18)

I am still thinking we can handle Surrogate by special casing that in the code and not including this whole data. 

As we discussed before Surrogate casing in specific range which we can just check in the code.

Reply:
CaseFolding.txt has _5 disjoint ranges_ for surrogates. There will be so many calculations for one surrogate that it will significantly slow down (I'd expect over 10x!) the processing of the string even if there is only one surrogate.

I assume that if consumers want to handle characters from Plane1, they should not be disappointed in .Net Core and use another product.

Potentially, the proposed code is the fastest, but we could use a three-level mapping that significantly reduces the size of the array.